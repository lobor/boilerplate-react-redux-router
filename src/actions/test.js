export function get(title) {
	return function (dispatch) {
		new Promise((resolve) => {
			dispatch( {
		    type: 'test',
		    title: title || 'hello'
		  })
		})
	}
};
