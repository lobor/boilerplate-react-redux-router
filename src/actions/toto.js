export function get(title = 'hello') {
	return {
    type: 'test',
    payload: { title }
	}
};
