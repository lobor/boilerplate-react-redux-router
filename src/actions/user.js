export function get(user = {}) {
	return {
    type: 'test',
    payload: { user }
	}
};

export function list(users = []) {
	return {
    type: 'test',
    payload: { users }
	}
};
