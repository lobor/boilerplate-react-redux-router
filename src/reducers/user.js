var initialState = {
	users: [
		{
			id: 1,
			lastName: 'Dupond',
			firstName: 'Lionel'
		},
		{
			id: 2,
			lastName: 'Tralfalgar',
			firstName: 'Nami'
		}
	],
	user: {}
};

export default function user(state = initialState, { type, payload }) {
  switch (type) {
    case 'LIST_USER':
      return state;
		case 'GET_USER':
			var user;
			state.users.forEach(function (el) {
				if (el.id === payload.id){
					user = el
				}
			})
      return Object.assign(state, { user: user });
    default:
      return state
  }
}
