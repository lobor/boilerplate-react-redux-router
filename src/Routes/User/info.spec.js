import React from 'react'
import renderer from 'react-test-renderer'
import { get } from 'actions/user'
import Info from './info'

const user = {
	id: 1,
	lastName: 'Dupond',
	firstName: 'Lionel'
}

describe('Test route User Info', () => {
	const tree = renderer.create(<Info location={{ params: { id: 1 } }} get={get} user={user} />).toJSON()

	test('render User', () => {
	  expect(tree).toMatchSnapshot()
	})
})
