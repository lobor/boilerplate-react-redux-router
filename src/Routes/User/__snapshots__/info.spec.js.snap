// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test route User Info render User 1`] = `
<div
  className="sc-htpNat bBHonn"
>
  <div
    className="sc-bxivhb lluzpC"
  >
    <label
      htmlFor="firstName"
    >
      First Name
    </label>
    <input
      className="sc-EHOje jbbhzM"
      id="firstName"
      name="firstName"
      type="text"
      value="Lionel"
    />
  </div>
  <div
    className="sc-bxivhb lluzpC"
  >
    <label
      htmlFor="lastName"
    >
      Last Name
    </label>
    <input
      className="sc-EHOje jbbhzM"
      id="lastName"
      name="lastName"
      type="text"
      value="Dupond"
    />
  </div>
</div>
`;
